File: app/domain/type/fizz_buzz_type_not_defined.go
package fizzbuzztype

import "fizzbuzz/domain/model"

// FizzBuzzTypeNotDefined 未定義タイプの実装（Null Objectパターン）
type FizzBuzzTypeNotDefined struct {
	FizzBuzzTypeBase
}

func (f FizzBuzzTypeNotDefined) Generate(number int) model.FizzBuzzValue {
	return model.NewFizzBuzzValue(number, "")
}

func (f FizzBuzzTypeNotDefined) String() string {
	return "未定義"
}

File: app/fizzbuzz-scala/src/main/scala/domain/types/FizzBuzzTypeNotDefined.scala
package domain.types

import domain.model.FizzBuzzValue

object FizzBuzzTypeNotDefined extends FizzBuzzType {
  override def generate(number: Int): FizzBuzzValue = {
    FizzBuzzValue(number, "")
  }
  
  override def toString: String = "未定義"
}
